engine/src/main/java/org/hibernate/search/engine/common/impl/IndexManagerBuildingStateHolder.java

```diff
@@ -163,22 +142,32 @@ private BackendInitialBuildState(
 				BackendImplementor<D> backend) {
 			this.backendName = backendName;
 			this.backendBuildContext = backendBuildContext;
+			this.backendPropertySource = backendPropertySource;
 			this.defaultIndexPropertySource =
 					EngineConfigurationUtils.getIndexDefaults( backendPropertySource );
 			this.backend = backend;
 		}
 
-		IndexManagerInitialBuildState<D> createIndexManagerBuildingState(
-				String indexName, boolean multiTenancyEnabled,
-				ConfigurationPropertySource indexPropertySource) {
-			ConfigurationPropertySource defaultedIndexPropertySource =
-					EngineConfigurationUtils.addIndexDefaults( indexPropertySource, defaultIndexPropertySource );
-			IndexManagerBuilder<D> builder = backend.createIndexManagerBuilder(
-					indexName, multiTenancyEnabled, backendBuildContext, defaultedIndexPropertySource
-			);
-			IndexSchemaRootNodeBuilder schemaRootNodeBuilder = builder.getSchemaRootNodeBuilder();
-			IndexedEntityBindingContext bindingContext = new IndexedEntityBindingContextImpl( schemaRootNodeBuilder );
-			return new IndexManagerInitialBuildState<>( backendName, indexName, builder, bindingContext );
+		public IndexManagerInitialBuildState<?> getIndexManagerBuildingState(
+				String indexName, boolean multiTenancyEnabled) {
+			IndexManagerInitialBuildState<?> state = indexManagerBuildStateByName.get( indexName );
+			if ( state == null ) {
+				ConfigurationPropertySource indexPropertySource =
+						EngineConfigurationUtils.getIndex( backendPropertySource, defaultIndexPropertySource, indexName );
+				ConfigurationPropertySource defaultedIndexPropertySource =
+						EngineConfigurationUtils.addIndexDefaults( indexPropertySource, defaultIndexPropertySource );
```

Comment:
I just noticed this: it's not necessary and should be removed. The line above already adds the default configuration as a fallback.